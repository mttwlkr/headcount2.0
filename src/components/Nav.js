import React, {Component} from 'react';
import '../styles/Nav.css';
import PropTypes from 'prop-types';

class Nav extends Component {
  constructor(props) {
    super(props);
    this.state = {
      search: ''
    }
    this.searchInput = this.searchInput.bind(this)
  }

  searchInput(e) {
    this.props.search(e.target.value)
    this.setState({
      search: e.target.value
    })
  }

  render() {
    return (
      <nav>
        <form>
          <input 
            type='text'
            value={this.state.search}
            onChange={this.searchInput}
          />
        </form>
      </nav>
    )
  }
}

Nav.propTypes = {
  search: PropTypes.func
}

export default Nav;