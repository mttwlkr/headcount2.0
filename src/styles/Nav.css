nav {
  margin: 10px auto;
  width: 40vw;
}



input {
  padding-left: 50px;
  border-radius: 25px;
  height: 50px;
  width: 100%;
  background-color: transparent;
  background-image: url('./search.png');
  background-repeat: no-repeat;
  background-size: 40px;
  background-position: 10px;
  box-sizing: border-box;
  color: #ffffff;
  font-size: 24px;
}

input:focus {
}

input::placeholder {
  color: #edeae0;
  font-size: 24px;
  /*padding-left: 40px;*/
}

