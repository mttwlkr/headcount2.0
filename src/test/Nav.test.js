import React from 'react';
import { shallow, mount } from 'enzyme';
import Nav from '../components/Nav';

describe('Nav', () => {
  it('should match the snapshot', () => {
    const wrapper = shallow(<Nav />);
    expect(wrapper).toMatchSnapshot()
  })

  it('should have a state of search set to empty string by default', () => {
    const wrapper = shallow(<Nav />);
    expect(wrapper.state().search).toEqual('')
  })

  it('should run the search prop method when searchInput is invoked', () => {
    const mockSearch = jest.fn();
    const mockEvent = {target: {value: 'example'}};
    const wrapper = shallow(<Nav search={mockSearch}/>)
    wrapper.instance().searchInput(mockEvent);
    expect(mockSearch).toHaveBeenCalled();
  })

  it('should update state on invoking searchInput', () => {
    const mockSearch = jest.fn();
    const mockEvent = {target: {value: 'example'}};
    const wrapper = shallow(<Nav search={mockSearch}/>);
    wrapper.instance().searchInput(mockEvent);
    expect(wrapper.state().search).toEqual(mockEvent.target.value)
    expect(wrapper.state().search).toEqual('example')
  })
})