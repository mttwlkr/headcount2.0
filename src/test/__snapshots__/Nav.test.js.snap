// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Nav />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <form>
        <input
                onChange={[Function]}
                type="text"
                value=""
        />
        <button
                type="submit"
        >
                Submit
        </button>
</form>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            onChange={[Function]}
            type="text"
            value=""
/>,
          <button
            type="submit"
>
            Submit
</button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "onChange": [Function],
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Submit",
            "type": "submit",
          },
          "ref": null,
          "rendered": "Submit",
          "type": "button",
        },
      ],
      "type": "form",
    },
    "type": "nav",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <form>
          <input
                    onChange={[Function]}
                    type="text"
                    value=""
          />
          <button
                    type="submit"
          >
                    Submit
          </button>
</form>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              onChange={[Function]}
              type="text"
              value=""
/>,
            <button
              type="submit"
>
              Submit
</button>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "onChange": [Function],
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Submit",
              "type": "submit",
            },
            "ref": null,
            "rendered": "Submit",
            "type": "button",
          },
        ],
        "type": "form",
      },
      "type": "nav",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
